core-bundle/public/ajax-form.js

```diff
@@ -0,0 +1,69 @@
+window.addEventListener('DOMContentLoaded', () => {
+    document.querySelectorAll('[data-ajax-form]').forEach(form => {
+        let triggeredButton = null;
+
+        function request(method, uri, body, callback) {
```

Comment:
```suggestion
        function request(method, form, body, callback) {
```

we need to pass the form here otherwise the reference is broken after the call to `form.replaceWith(newForm)`